### PR TITLE
frontend: Changes image for creation templates for openshift resources

### PR DESCRIPTION
**Pull request by amrutac**
Fixes https://jira.coreos.com/browse/CONSOLE-428

### CODE REVIEW

**Comment by spadgett** [comment]
Would it be easier to always use an image that works on both OpenShift and Kubernetes in the templates instead of trying to replace the image on OpenShift clusters? (It doesn't have to be hello-openshift... It just can't require root.)

**Comment by amrutac** [comment]
@spadgett: I didn't know if there was an image that would work for both set of templates. I can check around but if you are aware of one, let me know. Thanks!

**Comment by spadgett** [comment]
Anything that runs on OpenShift should run on Kubernetes. We could just switch to `openshift/hello-openshift` although it would be nice to keep real world examples like nginx or redis.

@bparees Do you have suggestions on good example images we can use? We have YAML templates for different types like deployments. For instance,

https://github.com/openshift/console/blob/master/frontend/public/components/deployment.jsx#L10-L28

The current images we use crash loop on OpenShift since they need root.

**Comment by bparees** [comment]
> @bparees Do you have suggestions on good example images we can use? We have YAML templates for different types like deployments. 

good examples of what?

images that work well on openshift as is would be:
1) our jenkins image
2) our various DB images (postgres, mysql, mongo, redis)
3) the hello-openshift image you noted

(1) and (2) generally require at least some env variable configuration to start up properly though.

**Comment by spadgett** [comment]
OK thanks @bparees.

@amrutac Thinking about this more, I'd suggest just sticking with hello-openshift. Then you can edit the image name in the YAML without having to remove a bunch of env vars. Plus it's really really small (<10MB), whereas some of these others aren't.

**Comment by amrutac** [comment]
@bparees @spadgett: Thanks! 

@spadgett: Updated images for templates.

**Comment by kans** [comment]
jenkins rebuild

File: frontend/public/components/deployment.jsx [hunk 1]
```diff
@@ -14,16 +14,16 @@ metadata:
...
-        app: nginx
+        app: hello-openshift
     spec:
       containers:
-      - name: nginx
-        image: nginx
+      - name: hello-openshift
+        image: openshift/hello-openshift
         ports:
         - containerPort: 80`);
```
Comment (spadgett) [inline frontend/public/components/deployment.jsx]: 6061

File: frontend/public/components/pod.jsx [hunk 2]
```diff
@@ -22,11 +22,11 @@ kind: Pod
...
   labels:
-    app: redis
+    app: hello-openshift
 spec:
   containers:
     - name: key-value-store
-      image: redis
+      image: openshift/hello-openshift
       ports:
         - containerPort: 6379`);
```
Comment (spadgett) [inline frontend/public/components/pod.jsx]: 6061

File: frontend/public/components/replicaset.jsx [hunk 3]
```diff
@@ -14,16 +14,16 @@ spec:
...
-        app: nginx
+        app: hello-openshift
     spec:
       containers:
-      - name: nginx
-        image: nginx
+      - name: hello-openshift
+        image: openshift/hello-openshift
         ports:
         - containerPort: 80`);
```
Comment (spadgett) [inline frontend/public/components/replicaset.jsx]: 6061

File: frontend/public/components/replication-controller.jsx [hunk 4]
```diff
@@ -15,16 +15,16 @@ metadata:
...
-        app: nginx
+        app: hello-openshift
     spec:
       containers:
-      - name: nginx
-        image: nginx
+      - name: hello-openshift
+        image: openshift/hello-openshift
         ports:
         - containerPort: 80`);
```
Comment (spadgett) [inline frontend/public/components/replication-controller.jsx]: 6061

File: frontend/public/components/job.jsx [hunk 5]
```diff
@@ -15,11 +15,11 @@ spec:
...
     metadata:
-      name: pi
+      name: hello-openshift
     spec:
       containers:
-      - name: pi
-        image: perl
+      - name: hello-openshift
+        image: openshift/hello-openshift
         command: ["perl",  "-Mbignum=bpi", "-wle", "print bpi(2000)"]
```
Comment (spadgett) [inline frontend/public/components/job.jsx]: Looks like this one runs OK on OpenShift. I'd leave it as it was.

File: frontend/public/components/cron-job.jsx [hunk 6]
```diff
@@ -17,7 +17,7 @@ spec:
         spec:
           containers:
           - name: hello
-            image: busybox
+            image: openshift/hello-openshift
```
Comment (spadgett) [inline frontend/public/components/cron-job.jsx]: This one runs OK, too. I'd leave it as is.

**Comment by spadgett** [comment]
Sorry port should be 8080 in all cases, not 6061

**Comment by amrutac** [comment]
@spadgett: Thanks! Updated, RFR

File: frontend/public/components/pod.jsx [hunk 7]
```diff
@@ -22,13 +22,13 @@ kind: Pod
 metadata:
   name: example
   labels:
-    app: redis
+    app: hello-openshift
 spec:
   containers:
     - name: key-value-store
```
Comment (spadgett) [inline frontend/public/components/pod.jsx]: Missed this last review, but probably should change container name here, too.

Comment (amrutac) [reply frontend/public/components/pod.jsx]: Fixed

Comment (spadgett) [reply frontend/public/components/pod.jsx]: Thanks, did you push the change?

Comment (amrutac) [reply frontend/public/components/pod.jsx]: OK, it's fixed now.

**Comment by spadgett** [comment]
I think we have a timing issue with the bot if the tag is added and the jobs haven't run :(

**Pull request merged**